base/fill_struct.go

```diff
@@ -0,0 +1,243 @@
+package base
```

Comment:
Fill struct is really good. So good that I think it needs to be it's own package. What do you think of moving the fill_struct code to `qri-io/dataset/fill`? The only exported function could be renamed from `FillStruct` to `Struct`, (package consumers would call `fill.Struct`.

This pattern has legs, and I think it's already forced a number of important conversations about adjacent interface decisions like paths vs. keys; case sensitivity; the role of fill struct in the serialization chain; etc. All of this could be documented better if `fill_struct` had it's own package.

Reply:
Discussed offline, I feel like fill_struct has some UI level concerns with how it handles case-sensitivity and unknown fields, that make it a bit more specific to qri the app than just dataset. Also, it could be used for config.Config, or other structs that are separate from dataset. It could be its own package, or a subpackage of `base`, which @b5 will probably do soon.